RFC-0026-logging-system.md

```diff
@@ -0,0 +1,103 @@
+# PyTorch Logging System
+
+## **Summary**
+Create a message logging system for PyTorch with the following requirements:
+
+* All errors, warnings, and other messages generated by PyTorch should be
+  emitted using the the logging system API
+
+* The APIs for emitting messages and changing settings should all be consistent
+  between C++ and Python
+
+* Offer different message severity levels, including at least the following:
+
+  - **Info**: Emits a message without creating a warning or error. By default,
+    this gets printed to stdout
+
+  - **Warning**: Emits a message as a warning. By default, this will turn into
+    a Python warning
+
+  - **Error**: Emits a message as an error. By default, this will turn into
+    a Python error
+
+  - TODO: Should we also have a **Fatal** severity for integration with
+    Meta's internal logging system? A fatal message terminates the program
+
+* Offer different classes of messages, including at least the following:
+
+  - **Default**: A catch-all message class
+
+  - **Nondeterministic**: Emitted when `torch.use_deterministic_algorithms(True)`
+    is set and a nondeterministic operation is called
+
+  - **Deprecated**: Emitted when a deprecated function is called
+
+  - **Beta**: Emitted when a beta feature is called. See
+    [PyTorch feature classifications](https://pytorch.org/blog/pytorch-feature-classification-changes/)
+
+  - **Prototype**: Emitted when a prototype feature is called. See
+    [PyTorch feature classifications](https://pytorch.org/blog/pytorch-feature-classification-changes/)
+
+* Continue using warning/error APIs that currently exist in PyTorch wherever
+  possible. For instance, `TORCH_CHECK`, `TORCH_WARN`, and `TORCH_WARN_ONCE`
+  should continue to be used in C++
+
+  - NOTE: These existing APIs don't currently have a concept of message classes,
```

Comment:
Being added or simply integrated in the sense that `TORCH_CHECh` is an RuntimeError, `TORCH_WARN` is a basic warning, `TORCH_CHECK_TYPE` is a TypeError, etc